tests/php/ORM/DataObjectSchemaTest.php

```diff
@@ -373,4 +399,76 @@ public function testWriteUniqueIndexes()
             AllIndexes::get()
         );
     }
+
+    public function provideHasOneComponent()
+    {
+        return [
+            [
+                'class' => WithRelation::class,
+                'component' => 'Relation',
+                'expected' => HasFields::class,
+            ],
+            [
+                'class' => WithRelation::class,
+                'component' => 'PolymorphicRelation',
+                'expected' => DataObject::class,
+            ],
+            [
+                'class' => WithRelation::class,
+                'component' => 'ArraySyntaxRelation',
+                'expected' => HasFields::class,
+            ],
+            [
+                'class' => WithRelation::class,
+                'component' => 'MultiReciprocalRelation',
+                'expected' => DataObject::class,
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider provideHasOneComponent
+     */
+
+    public function testHasOneComponent(string $class, string $component, string $expected)
+    {
+        $this->assertSame($expected, DataObject::getSchema()->hasOneComponent($class, $component));
+    }
+
+    public function provideHasOneComponentHandlesMultipleHasMany()
```

Comment:
```suggestion
    public function provideHasOneComponentHandlesMultipleHasMany(): array
```
Move provide method below the test method

Reply:
Done